src/config/mongoose_config_parser_toml.erl

```diff
@@ -607,13 +620,18 @@ riak_credentials([<<"password">>|_], V) -> [{password, b2l(V)}].
 -spec cassandra_option(path(), toml_value()) -> [option()].
 cassandra_option([<<"servers">>|_] = Path, V) -> [{servers, parse_list(Path, V)}];
 cassandra_option([<<"keyspace">>|_], KeySpace) -> [{keyspace, b2l(KeySpace)}];
-cassandra_option([<<"tls">>|_] = Path, Options) -> [{ssl, parse_section(Path, Options)}].
+cassandra_option([<<"tls">>|_] = Path, Options) -> [{ssl, parse_section(Path, Options)}];
+cassandra_option([<<"auth">>|_], Options) ->
+    Mod = b2a(maps:get(<<"module">>, Options)),
+    Args = lists:map(fun list_to_tuple/1, maps:get(<<"options">>, Options)),
+    [{auth, {Mod, Args}}].
```

Comment:
Can we do that using `parse_section` function to be consistent with the rest of the parser?

Reply:
No, because to do that we'd have to define all possible keys and create handlers for them, while here we don't know what the keys are going to be (unless we choose option two, see my long comment from this morning).